doc/sphinx/addendum/sprop.rst

```diff
@@ -0,0 +1,197 @@
+.. _sprop:
+
+SProp (proof irrelevant propositions)
+=====================================
+
+.. warning::
+
+   The status of strict propositions is experimental.
+
+This section describes the extension of |Coq| with definitionally
+proof irrelevant propositions (types in the sort :math:`\SProp`, also
+known as strict propositions). To use :math:`\SProp` you must pass
+``-allow-sprop`` to the |Coq| program.
+
+Basic constructs
+----------------
+
+The purpose of :math:`\SProp` is to provide types where all elements
+are convertible:
+
+.. coqtop::
+
+   Definition irrelevance (A:SProp) (P:A -> Prop) (x:A) (v:P x) (y:A) : P y := v.
+
+Since we have definitional η-expansion :ref:`eta-expansion` for
+functions, the property of being a type of definitionally irrelevant
+values is impredicative, and so is :math:`\SProp`:
+
+.. coqtop::
+
+   Check fun (A:Type) (B:A -> SProp) => (forall x:A, B x) : SProp.
+
+In order to keep conversion tractable, cumulativity for :math:`\SProp`
+is forbidden:
+
+.. coqtop::
+
+   Fail Check (fun (A:SProp) => A : Type).
+
+We can explicitly lift strict propositions into the relevant world by
+using a wrapping inductive type. The inductive stops definitional
+proof irrelevance from escaping.
+
+.. coqtop::
+
+   Inductive Box (A:SProp) : Prop := box : A -> Box A.
+   Arguments box {_} _.
+
+   Fail Check fun (A:SProp) (x y : Box A) => eq_refl : x = y.
+
+   Definition box_irrelevant (A:SProp) (x y : Box A) : x = y
+     := match x, y with box x, box y => eq_refl end.
+
+In the other direction, we can use impredicativity to "squash" a
+relevant type, making an irrelevant approximation.
+
+.. coqtop::
+
+  Definition iSquash (A:Type) : SProp
+    := forall P : SProp, (A -> P) -> P.
+  Definition isquash A : A -> iSquash A
+    := fun a P f => f a.
+  Definition iSquash_sind A (P : iSquash A -> SProp) (H : forall x : A, P (isquash A x))
+    : forall x : iSquash A, P x
+    := fun x => x (P x) (H : A -> P x).
+
+.. note::
+
+   Eliminators to strict propositions are called ``foo_sind``, in the
+   same way that eliminators to propositions are called ``foo_ind``.
+
+Or more conveniently (but equivalently)
+
+.. coqtop::
+
+  Inductive Squash (A:Type) : SProp := squash : A -> Squash A.
+
+Most inductives types defined in :math:`\SProp` are squashed types,
+i.e. they can only be eliminated to construct proofs of other strict
+propositions. The exceptions are empty types and unit types.
+
+.. coqtop::
+
+   Inductive sEmpty : SProp := .
+   Check sEmpty_rect.
+
+   Inductive sUnit : SProp := stt.
+   Check sUnit_rect.
```

Comment:
This is a case where the reader really needs to see the output of `coqtop`. Note that I'm surprised that the output is not shown by default when not passing an option. Anyways, changing `.. coqtop::` into `.. coqtop:: all` should fix this.

Reply:
@Zimmi48 I ended up removing relevant matches for irrelevant unit types (and the corresponding doc bit), is this comment still relevant?

Reply:
You mean you removed elimination from sUnit to Type?

Reply:
>You mean you removed elimination from sUnit to Type?

Yes. It would need to always reduce (so needing so special handling in reduction) and it's useless (the identity function works fine) so I just removed it.

Reply:
>is this comment still relevant?

I suppose not, but if you never intend to show the output of `.. coqtop::` you should not waste computational resources and use `.. coqdoc::` instead.